crates/kubelet/src/kubelet.rs

```diff
@@ -67,33 +68,39 @@ pub struct Status {
 pub struct Kubelet<P: 'static + Provider + Clone + Send + Sync> {
     provider: Arc<Mutex<P>>,
     kubeconfig: Configuration,
+    config: Config,
 }
 
 impl<T: 'static + Provider + Sync + Send + Clone> Kubelet<T> {
     /// Create a new Kubelet with a provider, a KubeConfig, and a namespace.
-    pub fn new(provider: T, kubeconfig: Configuration) -> Self {
+    pub fn new(provider: T, kubeconfig: Configuration, config: Config) -> Self {
         Kubelet {
             provider: Arc::new(Mutex::new(provider)),
             kubeconfig,
+            config,
         }
     }
 
     /// Begin answering requests for the Kubelet.
     ///
     /// This will listen on the given address, and will also begin watching for Pod
     /// events, which it will handle.
-    pub async fn start(&self, address: std::net::SocketAddr) -> Result<(), failure::Error> {
+    pub async fn start(&self) -> Result<(), failure::Error> {
         self.provider.lock().await.init().await?;
         let client = APIClient::new(self.kubeconfig.clone());
         // Create the node. If it already exists, "adopt" the node definition
-        create_node(&client, &self.provider.lock().await.arch()).await;
+        let mut conf = self.config.clone();
+        conf.arch = self.provider.lock().await.arch();
```

Comment:
Seems weird that we have an `arch` field that is normally set to "" that we have to remember to update with the arch from provider. Maybe passing around `arch` separately makes more sense?

Reply:
I couldn't figure out a nice way to do this originally. The problem is I don't want a user to have to know or set the arch manually (even with a constant). Is there a way I can do this cleanly?

Reply:
is `create_node` supposed to be public? i.e., will a user call it?

Reply:
`create_node` should not be public, so I'll go ahead and remove the `pub` modifier

Reply:
That module shouldn't be. I went ahead and removed it. I also just had the underlying functions in the node module accept arch as an argument and removed it from the config